src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -592,12 +575,14 @@ internal String ToString(TraceFormat traceFormat)
                             sb.Append(' ');
                             sb.AppendFormat(CultureInfo.InvariantCulture, inFileLineNum, fileName, sf.GetFileLineNumber());
                         }
-                    }
 
-                    if (sf.GetIsLastFrameFromForeignExceptionStackTrace())
-                    {
-                        sb.Append(Environment.NewLine);
-                        sb.Append(SR.Exception_EndStackTraceFromPreviousThrow);
+                        if (sf.GetIsLastFrameFromForeignExceptionStackTrace() && 
+                            !(declaringType?.Assembly.FullName.StartsWith(CoreLib.Name, StringComparison.Ordinal) ?? false) && // Skip EDI boundary for mscorlib
```

Comment:
Doing a direct comparison on the assembly object seems likely to be faster and more robust. Something like
static Assembly CorelibAssembly = typeof(object).Assembly?

Reply:
Done